### PR TITLE
Fix ASHRAE simple method for vertical surfaces

### LINKED ISSUE #7889: Fix ASHRAE Simple Convection Coefficient Initialization
Issue overview
--------------
The function for the setting the convection coefficient (HConvIn) on surfaces using the ASHRAE Simple algorithm currently does not attempt to calculate coefficient if the tilt of the surface is above 67.5 degrees (up to 90 degrees).  This results in surfaces falling within this range of tilt having a convection coefficient relying on the global initialization of HConvIn, which works in most cases, but will not be correct if the convection algorithm ever changes and sets HConvIn to a different value for the surface.

### Details
Some additional details for this issue (if relevant):
 - Platform: All
 - Version of EnergyPlus: 9.2

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by ajscimone**
Pull request overview
---------------------
 - Fixes #7889 
 - This pull request fixes the initialization of HConvIn, the convection coefficient, for ASHRAE Simple algorithm calculations. Prior to this pull request, the HConvIn returned from the ConvectionCoefficients namespace is a `NaN` for surfaced tilted above 67.5 degrees. This essentially covers the "vertical surface" cases fed into the ASHRAE Simple algorithm. After this pull request, HConvIn is set correctly for all cases, and unit tests have been added to ensure that surfaces which don't already have an HConvIn set are assigned a convection coefficient correctly, and that the CalcASHRAESimpleIntConvCoeff function returns the correct coefficient for each case defined by the algorithm.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by mitchute** [comment]
@ajscimone feel free to pull in develop again here. There's one minor merge conflict that isn't really a conflict.

I merged in develop and ran tests locally, but I can't push to your fork. I'll work on reviewing it once CI takes another pass here.

**Comment by ajscimone** [comment]
@mitchute I re-merged develop and saw the white space merge conflict you were referring to. Everything should be fixed and up to date now, just waiting for the CI.

File: src/EnergyPlus/ConvectionCoefficients.cc [hunk 1]
```diff
@@ -3831,62 +3830,70 @@ namespace ConvectionCoefficients {
...
-                return 2.281;
-            } else if (DeltaTemp * cosTilt == 0.0) { // Vertical Surface
-                return 3.076;
-            } else /*if (DeltaTemp * cosTilt > 0.0)*/ { // Tilted, Enhanced Convection
-                return 3.870;
+        if (std::abs(cosTilt) < 0.3827) {  // Vertical Surface
+            return 3.076;
+        }
+        else {
+            Real64 DeltaTempCosTilt = (Tamb - Tsurf)*cosTilt;
```
Comment (Myoldmopar) [inline src/EnergyPlus/ConvectionCoefficients.cc:3854]: A little whitespace around that asteriscus would've been nice but not a big deal.

File: src/EnergyPlus/ConvectionCoefficients.cc [hunk 2]
```diff
@@ -3831,62 +3830,70 @@ namespace ConvectionCoefficients {
...
+        //      | Horizontal Surface  | Positive  | -0.9238 to -1.0 (157.5 to 180 degrees)      | Negative         | Reduced         |       0.948 |
+        //      | Horizontal Surface  | Negative  | 0.9239 to 1.0 (0 to 22.5 degrees)           | Negative         | Reduced         |       0.948 |
+        //      | Horizontal Surface  | Negative  | -0.9239 to -1.0 (157.5 to 180 degrees)      | Positive         | Enhanced        |       4.040 |
+        //      | Tilted Surface      | Positive  | 0.3827 to 0.9239 (22.5 to 67.5 degrees)     | Positive         | Enhanced        |       3.870 |
+        //      | Tilted Surface      | Negative  | -0.3827 to -0.9239 (157.5 to 157.5 degrees) | Positive         | Enhanced        |       3.870 |
+        //      | Tilted Surface      | Negative  | 0.3827 to 0.9239 (22.5 to 67.5 degrees)     | Negative         | Reduced         |       2.281 |
+        //      | Tilted Surface      | Positive  | -0.3827 to -0.9239 (157.5 to 157.5 degrees) | Negative         | Reduced         |       2.281 |
+        //      +---------------------+-----------+---------------------------------------------+------------------+-----------------+-------------+
```
Comment (Myoldmopar) [inline src/EnergyPlus/ConvectionCoefficients.cc:3846]: In a function like this with a bunch of code paths, this is a great addition to help a developer understand what _should_ be happening, and also the limitations of the function.

File: tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc [hunk 3]
```diff
@@ -1242,3 +1242,139 @@ TEST_F(ConvectionCoefficientsFixture, CalcBeausoleilMorrisonMixedUnstableCeiling
...
+    DataSurfaces::Surface(1).CosTilt = 0;
+
+    DataHeatBalance::HConvIn.allocate(1);
+
+    CalcASHRAESimpleIntConvCoeff(1, 20.0, 30.0);
+
+    ConvectionCoefficient = DataHeatBalance::HConvIn(1);
+
+    EXPECT_EQ(ConvectionCoefficient, ExpectedCoefficient);
+}
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/ConvectionCoefficients.unit.cc:1380]: Great set of unit tests.  Kind of a shame that you missed one code path in all the possible configurations (line 3874 [here](http://energyplus.s3-website-us-east-1.amazonaws.com/coverage/2020-05-15-EnergyPlus-9bf737ca59-PR7890-x86_64-gcc-7.5UTCD/EnergyPlus/ConvectionCoefficients.cc.gcov.html)), but again, not going to hold this up for that, this is good work.

File: src/EnergyPlus/ConvectionCoefficients.cc [hunk 4]
```diff
@@ -3831,62 +3830,70 @@ namespace ConvectionCoefficients {
...
-                };
-            } else {
-                HConvIn(SurfNum) = CalcASHRAESimpleIntConvCoeff(SurfaceTemperature, ZoneMeanAirTemperature, Surface(SurfNum).CosTilt);
-            }
+        if (Surface(SurfNum).ExtBoundCond == DataSurfaces::KivaFoundation){
+            SurfaceGeometry::kivaManager.surfaceConvMap[SurfNum].in = [](double Tsurf, double Tamb, double, double, double cosTilt) -> double {
+              return CalcASHRAESimpleIntConvCoeff(Tsurf, Tamb, cosTilt);
+            };
+        }
+        else {
```
Comment (Myoldmopar) [inline src/EnergyPlus/ConvectionCoefficients.cc:3891]: This else style is not the preferred form.  `} else {` is preferred.

File: src/EnergyPlus/ConvectionCoefficients.cc [hunk 5]
```diff
@@ -49,7 +49,6 @@
 #include <algorithm>
 #include <cassert>
 #include <cmath>
-#include <limits>
```
Comment (Myoldmopar) [inline src/EnergyPlus/ConvectionCoefficients.cc:52]: Removing includes :+1:

**Review by Myoldmopar — APPROVED** [review]
I left a couple minor comments, but overall this is good to go.  I'm building locally with very latest develop to verify the fix and this will be able to drop in.

**Comment by Myoldmopar** [comment]
CI is generally really happy.  I'm not worried about the Windows bot, it looks to be due to the branch being behind develop.  I'm running it locally to verify the build and fix.

**Comment by Myoldmopar** [comment]
No defect file, but built and ran unit test suite locally.  Good to go.  Merging.

**Pull request merged**